services/cd-service/pkg/service/batch.go

```diff
@@ -92,12 +93,37 @@ func ValidateApplication(
 	return nil
 }
 
+func (d *BatchServer) checkUserPermissions(user *auth.User, act *api.CreateEnvironmentLockRequest, application, action string) error {
+	if !d.RBACConfig.DexEnabled {
+		return nil
+	}
+	groups, err := d.Repository.State().GetEnvironmentConfigs()
+	if err != nil {
+		return err
+	}
+	group := act.Environment
+	temp := groups[act.Environment].EnvironmentGroup
+	if temp != nil {
+		group = *temp
+	}
+	_, permissionsExist := d.RBACConfig.Policy[fmt.Sprintf("p,%s,%s,%s,%s,allow", user.DexAuthContext.Role, application, action, fmt.Sprintf("%s:%s", act.Environment, group))]
+	if !permissionsExist {
+		return status.Errorf(codes.PermissionDenied, "user does not have permissions to create an environment lock")
```

Comment:
please return more details here, like which line you were looking for in the rbacconfig.Policy.